objects/src/transaction/created_notes.rs

```diff
@@ -0,0 +1,157 @@
+        let commitment = generate_created_notes_stub_commitment(&notes);
+        Self { notes, commitment }
+    }
+
+    // PUBLIC ACCESSORS
+    // --------------------------------------------------------------------------------------------
+    /// Returns a reference to the vector of [NoteStub]s.
+    pub fn notes(&self) -> &[NoteStub] {
+        &self.notes
+    }
+
+    /// Returns the commitment to the created notes.
+    pub fn commitment(&self) -> Digest {
+        self.commitment
+    }
+}
+
+impl TryFromVmResult for CreatedNotes {
+    type Error = TransactionResultError;
+
+    fn try_from_vm_result(
+        stack_outputs: &StackOutputs,
+        _advice_stack: &[Felt],
+        advice_map: &BTreeMap<[u8; 32], Vec<Felt>>,
+        _merkle_store: &MerkleStore,
+    ) -> Result<Self, Self::Error> {
+        const CREATED_NOTES_COMMITMENT_WORD_IDX: usize = 0;
+
+        let created_notes_commitment: Word =
+            stack_outputs.stack()[CREATED_NOTES_COMMITMENT_WORD_IDX * WORD_SIZE
+                ..(CREATED_NOTES_COMMITMENT_WORD_IDX + 1) * WORD_SIZE]
+                .iter()
+                .rev()
+                .map(|x| Felt::from(*x))
+                .collect::<Vec<_>>()
+                .try_into()
+                .expect("word size is correct");
+        let created_notes_commitment: Digest = created_notes_commitment.into();
+
+        let created_notes_data = advice_map
+            .get(&created_notes_commitment.as_bytes())
+            .ok_or(TransactionResultError::CreatedNoteDataNotFound)?;
+
+        let mut created_notes = Vec::new();
+        let mut created_note_ptr = 0;
+        while created_note_ptr < created_notes_data.len() {
+            let note_stub: NoteStub = created_notes_data[created_note_ptr..]
+                .try_into()
+                .map_err(TransactionResultError::CreatedNoteDataInvalid)?;
+            created_notes.push(note_stub);
+            created_note_ptr += (NOTE_MEM_SIZE as usize) * WORD_SIZE;
+        }
+
+        let created_notes = Self::new(created_notes);
+        if created_notes_commitment != created_notes.commitment() {
+            return Err(TransactionResultError::CreatedNotesCommitmentInconsistent(
+                created_notes_commitment,
+                created_notes.commitment(),
+            ));
+        }
+
+        Ok(created_notes)
+    }
+}
+
+// CREATED NOTE INFO
+// ================================================================================================
+
+/// Holds information about a note that was created by a transaction.
+/// Contains:
+/// - note_hash: hash of the note that was created
+/// - note_metadata: metadata of the note that was created. Metadata is padded with ZERO such that
+///   it is four elements in size (a word). The metadata includes the following elements:
+///     - sender
+///     - tag
+///     - ZERO
+///     - ZERO
+pub struct CreatedNoteInfo {
+    note_hash: Digest,
+    note_metadata: NoteMetadata,
+}
```

Comment:
If we could come up with a more generic name for `CreatedNoteInfo`, we could define `NotStub` to build on it. For example:
```Rust
pub struct NoteStub {
    info: CreatedNoteInfo,
    recipient: Digest,
    vault: NoteVault,
}
```

Reply:
I've gone for `NoteEnvelope` however I also think `NoteBulletin` is a worthy consideration.  Let me know your thoughts.

Reply:
I like `NoteEnvelope` better - so, let's keep it. If we come up with something better, should be easy enough to rename.